src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -761,11 +761,16 @@ public DatasetSummaryModel updateTags(UUID id, TagUpdateRequestModel tagUpdateRe
 
   public String enableInheritSteward(UUID datasetId, AuthenticatedUserRequest userReq) {
     String description = "Enable InheritSteward for dataset " + datasetId;
+    var custodianEmail =
+        iamService
+            .retrievePolicyEmails(userReq, IamResourceType.DATASET, datasetId)
+            .get(IamRole.CUSTODIAN);
```

Comment:
Oh nice, this allows us to avoid adding an additional step to populate this value in the flight map.